backend/src/test/java/de/neuefische/backend/service/EventHubUserDetailServiceTest.java

```diff
@@ -0,0 +1,108 @@
+    GenerateUUIDService generateUUIDService = mock(GenerateUUIDService.class);
+    PasswordEncoder passwordEncoder = mock(PasswordEncoder.class);
+    EventHubUserDetailService eventHubUserDetailService = new EventHubUserDetailService(userRepository, passwordEncoder, generateUUIDService);
+
+    @Test
+    void testGetUserPreferredCategories_case_preferred_categories_set() {
+        // given
+        String username = "user@event.hub";
+        List<SimpleGrantedAuthority> roles = List.of(new SimpleGrantedAuthority("ROLE_USER"));
+        List<EventCategory> preferredCategories = List.of(EventCategory.MUSIC, EventCategory.SPORTS);
+        EventHubUser user = new EventHubUser("1", "user@event.hub", "123", roles, preferredCategories);
+        when(userRepository.findEventHubUserByUsername(username)).thenReturn(Optional.of(user));
+        List<String> expectedCategories = List.of("MUSIC", "SPORTS");
+
+        // when
+        List<String> userPreferredCategories = eventHubUserDetailService.getUserPreferredCategories(username);
+
+        // then
+        assertEquals(expectedCategories, userPreferredCategories);
+    }
+
+    @Test
+    void testGetUserPreferredCategories_case_user_not_found() {
+        // given
+        String username = "user@event.hub";
+
+        // when & then
+        assertThrows(UsernameNotFoundException.class, () -> eventHubUserDetailService.getUserPreferredCategories(username));
+    }
+
+    @Test
+    void testGetUserPreferredCategories_case_no_preferred_categories_present() {
+        // given
+        String username = "user@event.hub";
+        List<SimpleGrantedAuthority> roles = List.of(new SimpleGrantedAuthority("ROLE_USER"));
+        EventHubUser user = new EventHubUser("1", username, "123", roles, null);
+        when(userRepository.findEventHubUserByUsername(username)).thenReturn(Optional.of(user));
+
+        // when
+        List<String> userPreferredCategories = eventHubUserDetailService.getUserPreferredCategories(username);
+
+        // then
+        assertTrue(userPreferredCategories.isEmpty());
+    }
+
+    @Test
+    void testUpdateUserPreferredCategories_case_no_existing_categories() {
+        // given
+        String username = "user@event.hub";
+        List<EventCategory> categories = List.of(EventCategory.MUSIC, EventCategory.SPORTS);
+        UserPreferredCategoriesDTO userPreferredCategories = new UserPreferredCategoriesDTO(username, categories);
+        List<SimpleGrantedAuthority> roles = List.of(new SimpleGrantedAuthority("ROLE_USER"));
+        EventHubUser user = new EventHubUser("1", username, "123", roles, null);
+        when(userRepository.findEventHubUserByUsername(username)).thenReturn(Optional.of(user));
+
+        // when
+        EventHubUserDTO updatedUserPreferredCategories = eventHubUserDetailService.updateUserPreferredCategories(userPreferredCategories);
+
+        // then
+        assertEquals(username, updatedUserPreferredCategories.getUsername());
```

Comment:
Usually you have one assertEquals per tests. This looks like 3 tests in one. Why do you compare object properties and not the objects directly? Maybe create an expected object to simplify